doc/user/content/serve-results/s3.md

```diff
@@ -0,0 +1,231 @@
+               ],
+               "Resource": "arn:aws:s3:::<bucket>/<prefix>/*"
+           },
+           {
+               "Effect": "Allow",
+               "Action": [
+                   "s3:ListBucket"
+               ],
+               "Resource": "arn:aws:s3:::<bucket>",
+               "Condition": {
+                   "StringLike": {
+                       "s3:prefix": [
+                           "<prefix>/*"
+                       ]
+                   }
+               }
+           }
+       ]
+   }
+   ```
+
+1. Click **Next**.
+
+1. Enter a name for the policy, and click **Create policy**.
+
+### Create an IAM role
+
+Next, you must attach the policy you just created to a Materialize-specific
+[IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html).
+
+1. Navigate to **AWS Services**, then **AWS IAM**.
+
+1. In the **IAM Dashboard**, click **Roles**, then **Create role**.
+
+1. In **Trusted entity type**, select **Custom trust policy**, and copy and
+   paste the policy below.
+
+   ```json
+   {
+       "Version": "2012-10-17",
+       "Statement": [
+           {
+               "Effect": "Allow",
+               "Principal": {
+                   "AWS": "arn:aws:iam::664411391173:role/MaterializeConnection"
+               },
+               "Action": "sts:AssumeRole",
+               "Condition": {
+                   "StringEquals": {
+                       "sts:ExternalId": "mz_00000000-0000-0000-0000-000000000000_u0"
+                   }
+               }
+           }
+       ]
+   }
+   ```
+
+   Materialize **always uses the provided IAM principal** to assume the role, and
+   also generates an **external ID** which uniquely identifies your AWS connection
+   across all Materialize regions (see [AWS connection permissions](/sql/create-connection/#aws-permissions)).
+   For now, you'll set this ID to a dummy value; later, you'll update it with
+   the unique identifier for your Materialize region.
+
+1. Click **Next**.
+
+1. In **Add permissions**, select the IAM policy you created in [Create an IAM policy](#create-an-iam-policy),
+   and click **Next**.
+
+1. Enter a name for the role (e.g. `MZS3Exporter`), and click **Create role**.
+
+1. Click **View role** to see the role summary page, and note down the
+   role **ARN**. You will need it in the next step to create an AWS connection in
+   Materialize.
+
+## Step 2. Create a connection
+
+1. In the [SQL Shell](https://console.materialize.com/), or your preferred SQL
+   client connected to Materialize, create an [AWS connection](/sql/create-connection/#aws):
+
+   ```sql
+   CREATE CONNECTION aws_connection
+      TO AWS (ASSUME ROLE ARN = 'arn:aws:iam::400121260767:role/MZS3Exporter');
+   ```
+
+1. Retrieve the external ID for the connection:
+
+   ```sql
+   SELECT awsc.id, external_id
+    FROM mz_internal.mz_aws_connections awsc
+    JOIN mz_connections c ON awsc.id = c.id
+    WHERE c.name = 'aws_connection';
```

Comment:
It'd be nice to have an easier way to immediately access these values. In Snowflake, you can retrieve these details by typing `DESCRIBE STAGE`, though `DESCRIBE` is not a PostgreSQLism (it comes from Oracle and MySQL).

Reply:
Dedicated connections page in the console would be where I would start! We could add a `SHOW AWS CONNECTION` pretty easily too. But building out a dedicated console flow would probably be more impactful.

Reply:
`SHOW` is a bit of an awkward fit here. But agreed that this is the sort of feature (and details) that'd be really great to support in the console.

Reply:
What's awkward about it? Seems pretty clean:

```
SHOW AWS CONNECTIONS LIKE 'aws_connection'
```

Is it just the verb "show" instead of "describe"?

Reply:
Yeah, though it can just be the fact that we don't support `SHOW` for specific connection types yet!